src/main.cpp

```diff
@@ -99,6 +103,9 @@ int main(int argc, char *argv[])
             case 'M': // major frame start sync message
                 mf_sync_message = optarg;
                 break;
+            case 't': // scheduler timeout
+                scheduler_timeout = std::chrono::milliseconds{std::stoll(optarg)};
+                break;
```

Comment:
Quote a lot of code for functionality, which can probably be implemented using the `timeout` command. Is there anything more in this code that what `timeout` can do?

Reply:
Yes, `timeout` is not aware of the DEmOS internal setup and process initialization. This allows the user to reliably say "initialize everything and then run the scheduler for X milliseconds before stopping", with the knowledge that the time will be synchronized with the internal scheduler time (so the user knows exactly how many windows will be executed before stopping, etc).